contracts/amm/WTokenVault.sol

```diff
@@ -0,0 +1,261 @@
+pragma solidity 0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin/contracts/token/ERC1155/IERC1155.sol";
+import "../proxy/Proxiable.sol";
+import "./IMinterAmm.sol";
+import "../series/ISeriesController.sol";
+import "../series/SeriesLibrary.sol";
+import "../series/IPriceOracle.sol";
+import "../series/IVolatilityOracle.sol";
+import "../configuration/IAddressesProvider.sol";
+import "./IAmmDataProvider.sol";
+import "./IWTokenVault.sol";
+
+import "hardhat/console.sol";
+
+contract WTokenVault is OwnableUpgradeable, Proxiable, IWTokenVault {
+    /// @dev The address for the AddressesProvider
+    IAddressesProvider addressesProvider;
+
+    /// locked wTokens by seriesId
+    mapping(address => mapping(uint64 => uint256)) public lockedWTokens;
+
+    /// total supply of pool shares by expirationId
+    mapping(address => mapping(uint256 => uint256)) public lpSharesSupply;
+
+    /// balance of shares for each LP in expiration pool
+    mapping(address => mapping(uint256 => mapping(address => uint256)))
+        public lpShares;
+
+    /// locked collateral by expirationId
+    mapping(address => mapping(uint256 => uint256)) public lockedCollateral;
+
+    function initialize(IAddressesProvider _addressesProvider)
+        external
+        initializer
+    {
+        addressesProvider = _addressesProvider;
+        __Ownable_init();
+    }
+
+    function getWTokenBalance(address poolAddress, uint64 seriesId)
+        external
+        view
+        override
+        returns (uint256)
+    {
+        return lockedWTokens[poolAddress][seriesId];
+    }
+
+    struct LocalVars {
+        uint256 expirationIdMin;
+        uint256 expirationIdMax;
+        uint256 underlyingPrice;
+        uint256 volatility;
+        uint64[] allSeries;
+        uint256[] allExpirations;
+        uint256[] lockedValue;
+        uint256[] poolValue;
+    }
+
+    /// Lock active wTokens grouped by expiration
+    /// Assign number of shares to the locking address
+    function lockActiveWTokens(
```

Comment:
Do we want non reentrant on this?

Reply:
no need, because there's no token movement here, only accounting. The AMM has nonReentrant on the method that calls this